services/kvstore/src/main/java/tech/pegasys/pantheon/services/kvstore/RocksDbKeyValueStorage.java

```diff
@@ -81,6 +232,22 @@ private RocksDbKeyValueStorage(
           metricsSystem.createTimer(
               MetricCategory.ROCKSDB, "commit_latency_seconds", "Latency for commits to RocksDB.");
 
+      for (final TickerType ticker : TICKERS) {
+        final String promCounterName = ticker.name().toLowerCase();
+        metricsSystem.createLongGauge(
+            ROCKSDB_STATS,
+            promCounterName,
+            "RocksDB reported statistics for " + ticker.name(),
+            () -> stats.getTickerCount(ticker));
+      }
+
+      if (metricsSystem instanceof PrometheusMetricsSystem) {
+        for (final HistogramType histogram : HISTOGRAMS) {
+          ((PrometheusMetricsSystem) metricsSystem)
+              .addCollector(ROCKSDB_STATS, histogramToCollector(histogram));
+        }
+      }
```

Comment:
Can you wrap all of this new logic up in a utility class in the metrics package?  That way we can reuse this for other RocksDB instances as needed.  It looks like you could create a `RocksDBStats` class with methods `getStatistics()` and `registerMetrics(MetricsSystem metrics, MetricsCategory category)`.

Reply:
If we keeps the statistics object in rocksdb and treat it like a part of the options then we can have `RocksDBStats` have just static methods.